Kitodo-API/src/main/java/org/kitodo/config/Config.java

```diff
@@ -11,141 +11,197 @@
+     *            not exist or exception occurred
+     * @return value for the requested key, or {@code defaultIfNull} if not
+     *         found
      */
-    public static String getParameter(String parameter, String defaultIfNull) {
+    public static String getParameter(String key, String defaultValue) {
         try {
-            return getConfig().getString(parameter, defaultIfNull);
+            return getConfig().getString(key, defaultValue);
         } catch (ConversionException e) {
-            return defaultIfNull;
+            logger.catching(e);
+            logger.warn("Configuration found in kitodo_config.properties for key {} is defined, but not a String!",
+                key);
+            return defaultValue;
         }
     }
 
     /**
-     * Request boolean parameter from configuration, default if missing: false.
-     * 
-     * @param inParameter
-     *            name of parameter in config file
-     * @return Parameter as boolean
+     * Returns the selected boolean parameter from the configuration file. If no
+     * such parameter exists, or the value cannot be parsed to {@code boolean},
+     * returns {@code false}.
+     *
+     * @param key
+     *            key whose value is to be returned
+     * @return boolean value for the requested key, or {@code false} if not
+     *         found or not parsing
      */
-    public static boolean getBooleanParameter(String inParameter) {
-        return getBooleanParameter(inParameter, false);
+    public static boolean getBooleanParameter(String key) {
+        return getBooleanParameter(key, false);
     }
 
     /**
-     * Request boolean parameter from configuration.
+     * Returns the selected boolean parameter from the configuration file. If no
+     * such parameter exists, or the value cannot be parsed to {@code boolean},
+     * returns the provided default value.
      *
-     * @param parameter
-     *            name of parameter in config file
-     * @param defaultIfNull
-     *            default value in case parameter taken from config file is null or
-     *            exception occurred
-     * @return Parameter as boolean
+     * @param key
+     *            key whose value is to be returned
+     * @param defaultValue
+     *            default value in case parameter taken from config file does
+     *            not exist or exception occurred
+     * @return boolean value for the requested key, or {@code defaultIfNull} if
+     *         not found or not parsing
      */
-    public static boolean getBooleanParameter(String parameter, boolean defaultIfNull) {
+    public static boolean getBooleanParameter(String key, boolean defaultValue) {
         try {
-            return getConfig().getBoolean(parameter, defaultIfNull);
+            return getConfig().getBoolean(key, defaultValue);
         } catch (ConversionException e) {
-            return defaultIfNull;
+            logger.catching(e);
```

Comment:
Two different loggers in one catch block?

Reply:
This was there already. I only added logging of caught exception. Because as to coding guidelines, exception being caught must be logged.

Reply:
Is it necessary to double log it? I think one logger on level error is enough.

Reply:
same here